File: 2tip/2tip_web/cw4-forms/Models/Person.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace cw4_forms.Models;

public class Person
{
    [DisplayName("Podaj fajne imię")]
    [Required(ErrorMessage = "Imię jest wymagane")]
    public string? FirstName { get; set; }

    [DisplayName("Podaj nazwisko")]
    [Required(ErrorMessage = "Nazwisko jest wymagane")]
    public string? LastName { get; set; }

    [DisplayName("Podaj wiek")]
    [Required(ErrorMessage = "Wiek jest wymagane")]
    [Range(0, 120, ErrorMessage = "Wiek musi być z zakresu 0-120")]
    public int? Age { get; set; }
}

File: 2tip/2tip_cpp/cw21_dziedz/Person.cpp
#include "Person.h"
#include <iostream>
Person::Person():firstname("noname"),lastname("noname"),age(0) {
	std::cout << "Person()" << std::endl;
}
Person::Person(std::string firstname, std::string lastname, int age)
:firstname(firstname), lastname(lastname), age(age)
{
	std::cout << "Person(z argumentami)" << std::endl;
}

std::ostream& operator<<(std::ostream& os, const Person& p) {
	os << "Imie: " << p.firstname << " Nazwisko: " << p.lastname
	<< " Wiek: " << p.age<<std::endl;
	return os;
}